scripts/importing-Exporting-Config.js

```diff
@@ -0,0 +1,508 @@
+            {
+                var curr_loc_Strg_key = loc_Strge_Keys[k];
+                if (!curr_loc_Strg_key.startsWith(loc_Strg_Check)) continue; //we know curr key has data on selected team for curr event
+
+                if (curr_loc_Strg_key.includes('Match Data')) //add to match_Obj
+                {
+                    var match_Data_Arr = JSON.parse(localStorage[curr_loc_Strg_key]); //is an array of smaller element/frequency object pairs
+                    match_Obj_Value.push(match_Data_Arr);
+                }
+
+                if (curr_loc_Strg_key.includes('pit')) //add to pit_Obj
+                {
+                    var pit_Data_Arr = JSON.parse(localStorage[curr_loc_Strg_key]); //is an array of smaller Q/A object pairs
+                    pit_Obj_Value = pit_Data_Arr;
+                }
+            }
+
+            //set up match/pit objects
+            match_Obj[match_Obj_Name] = match_Obj_Value;
+            pit_Obj[pit_Obj_Name] = pit_Obj_Value;
+
+            //add match/pit objects to team object
+            team_Obj_Value.push(match_Obj);
+            team_Obj_Value.push(pit_Obj);
+
+            //add team object to event object
+            team_Obj[team_Obj_Name] = team_Obj_Value;
+            event_Obj_Value.push(team_Obj);
+        }
+
+        //push event obj to total output array
+        event_Obj[event_Obj_Name] = event_Obj_Value;
+        total_Output_Arr.push(event_Obj);
+    }
+
+    return total_Output_Arr;
+}
+
+
+
+
+// Importing Season config \/\/\/
+function readImportFile()
+{
+    var file = $('#importFile').prop('files')[0];
+
+    var fr = new FileReader();
+    fr.onload = function(e) {
+        var content = e.target.result;
+        var obj = JSON.parse(content);
+
+        if (file.name.endsWith('.rcubedscoutconfig'))
+        {
+            var continueOrNot = getImportModalChoice();
+            if (continueOrNot.includes('Yes')) //user clicked yes
+            {
+                closeImportModal();
+                processImportData(obj);
+                setCookie(import_Modal_Cookie_Name(),'',750); // reset cookie for new cycle
+            } else if (continueOrNot.includes('No')) //user clicked no
+            {
+                closeImportModal();
+                setCookie(import_Modal_Cookie_Name(),'',750); // reset cookie for new cycle
+            } else if (continueOrNot.trim().length == 0) //user hasn't selected yet or its the first time opening the modal
+            {
+                showImportModal(obj);
+            }
+            resetModalInputs();
+        } else
+        {
+            $('#importConfirmation').text('No ".rcubedscoutconfig" file selected.');
+        }
+    }
+    fr.readAsText(file);
+}
+
+function processImportData(obj) //goes through inner objects and assigns values to cookies
+{
+    var keys = Object.keys(obj);
+    var seasonList = obj[keys[0]]; //seasons at top of import config file
```

Comment:
this assumes the seasons list is at the top. JSON is not considered to be ordered, and different browsers may or may not include them in that order. Better to re-use the constant from my earlier comment.